[Bulk] Broken Link fixes (#878)
MicrosoftDocs/biztalk-docs: biztalk/technical-guides/post-configuration-database-optimizations2.md

-   Database auto-growth should be set to a fixed number of megabytes instead of to a percentage (specify file growth **In Megabytes**). This should be done so that, if auto-growth occurs, it does so in a measured fashion. This reduces the likelihood of excessive database growth. The growth increment for BizTalk Server databases should generally be no lower than 100 MB.

## Pre-size BizTalk Server databases to appropriate size with multiple data files

>  When SQL Server increases the size of a file, it must first initialize the new space before it can be used. This is a blocking operation that involves filling the new space with empty pages. SQL Server 2005 or later running on Windows Server 2003 or later supports “instant file initialization.” This can greatly reduce the performance impact of a file growth operation. For more information, see [Database File Initialization](https://go.microsoft.com/fwlink/?LinkId=132063) in the SQL Server books online. This topic provides steps for enabling instant file initialization.

Reviewer: ```suggestion
When SQL Server increases the size of a file, it must first initialize the new space before it can be used. This is a blocking operation that involves filling the new space with empty pages. SQL Server 2005 or later running on Windows Server 2003 or later supports instant file initialization. This capability can greatly reduce the performance impact of a file growth operation. For more information, see [Database Instant File Initialization](/sql/relational-databases/databases/database-instant-file-initialization), which provides steps for enabling instant file initialization.
```

```

## Move the Backup BizTalk Server output directory to a dedicated LUN

>  Move the Backup BizTalk (Full and Log backup) output directory to a dedicated LUN, and then edit the Backup BizTalk Server job to point to the dedicated LUN. Moving the Backup BizTalk Server output directory to a dedicated LUN will reduce disk I/O contention when the job is running by writing to a different disk than the job is reading from. For more information about configuring the Backup BizTalk Server job see [How to Configure the Backup BizTalk Server Job](../core/how-to-configure-the-backup-biztalk-server-job.md) in BizTalk Server 2010 help.

Reviewer: ```suggestion
1. Move the output directory named **Backup BizTalk (Full and Log backup)** to a dedicated LUN.

1. Edit the job named **Backup BizTalk Server** to point to the dedicated LUN.

The first step reduces disk I/O contention when the job is running by writing to a disk that differs from where the job is reading. For more information, see [Configure the Backup BizTalk Server Job](/biztalk/core/how-to-configure-the-backup-biztalk-server-job).
```

## Verify that the BizTalk Server SQL Agent Jobs are running

>  BizTalk Server includes several SQL Server Agent jobs that perform important functions to keep your servers operational and healthy. You should monitor the health of these jobs and ensure they are running without errors. One of the most common causes of performance problems in BizTalk Server is the BizTalk Server SQL Agent Jobs are not running, which in turn can cause the MessageBox and Tracking databases to grow unchecked. Follow these steps to ensure the BizTalk Server SQL Agent Jobs are running without problems:

Reviewer: ```suggestion
In BizTalk Server, several SQL Server Agent jobs perform important functions that keep your servers operational and healthy. Make sure that you monitor the health of these jobs and that they run without errors. One common cause of BizTalk Server performance problems is that when the BizTalk Server SQL Agent Jobs aren't running, the MessageBox and Tracking databases can grow unchecked. To make sure that the BizTalk Server SQL Agent Jobs run without problems, follow these steps:
```

## Configure Purging and Archiving of Tracking Data

>  Follow these steps to ensure that purging and archiving of tracking data is configured correctly:

Reviewer: ```suggestion
To make sure that you correctly set up purging and archiving tracking data, follow these steps:
```

> 1.  Ensure the SQL Agent job “DTA Purge and Archive” is properly configured, enabled, and successfully completing. For more information, see [How to Configure the DTA Purge and Archive Job](../core/how-to-configure-the-dta-purge-and-archive-job.md) in the BizTalk Server documentation.

Reviewer: ```suggestion
1. Make sure that the SQL Agent job named **DTA Purge and Archive** is properly configured, enabled, and successfully completing.

   For more information, see [Configure the DTA Purge and Archive Job](/biztalk/core/how-to-configure-the-dta-purge-and-archive-job).
```

> 2.  Ensure the job is able to purge the tracking data as fast as the incoming tracking data is generated. For more information, see [Measuring Maximum Sustainable Tracking Throughput](../core/measuring-maximum-sustainable-tracking-throughput.md) in the BizTalk Server documentation.

Reviewer: ```suggestion
2. Make sure that the job can purge the tracking data as fast as the incoming tracking data is generated.

   For more information, see [Measuring Maximum Sustainable Tracking Throughput](/biztalk/core/measuring-maximum-sustainable-tracking-throughput).
```

> 3.  Review the soft purge and hard purge parameters to ensure you are keeping data for the optimal length of time. For more information, see [Archiving and Purging the BizTalk Tracking Database](../core/archiving-and-purging-the-biztalk-tracking-database.md) in the BizTalk Server documentation.

Reviewer: ```suggestion
3. Review the soft purge and hard purge parameters to ensure you are keeping data for the optimal length of time.

   For more information, see [Archive and Purge the BizTalk Tracking Database](/biztalk/core/archiving-and-purging-the-biztalk-tracking-database).
```

> 4.  If you only need to purge the old data and do not need to archive it first, change the SQL Agent job to call the stored procedure “dtasp_PurgeTrackingDatabase.” For more information, see [How to Purge Data from the BizTalk Tracking Database](../core/how-to-purge-data-from-the-biztalk-tracking-database.md) in the BizTalk Server documentation.

Reviewer: ```suggestion
4. If you only need to purge the old data and don't need to archive first, change the SQL Agent job to call the stored procedure named **dtasp_PurgeTrackingDatabase**.

   For more information, see [Purge Data from the BizTalk Tracking Database](/biztalk/core/how-to-purge-data-from-the-biztalk-tracking-database).
```

  Separating the BizTalk MessageBox and BizTalk Tracking databases and separating the database files and transaction log files on different physical disks are considered best practices for reducing disk I/O contention. Try to spread the disk I/O across as many physical spindles as possible. You can also reduce disk I/O contention by placing the BizTalk Tracking database on a dedicated SQL Server; however, you should still follow the practices above with regards to separating data files and transaction log files.

## Optimize filegroups for the BizTalk Server databases

>  Follow the steps in [Optimizing Filegroups for the Databases2](../technical-guides/optimizing-filegroups-for-the-databases2.md) and the BizTalk Server Database Optimization white paper to create additional filegroups and files for the BizTalk Server databases. This will greatly increase the performance of the BizTalk Server databases from a single disk configuration.

Reviewer: Whitepaper doesn't exist:

```suggestion
 Follow the steps in [Optimizing Filegroups for the Databases2](../technical-guides/optimizing-filegroups-for-the-databases2.md) and [Optimizing Database Performance](/biztalk/technical-guides/optimizing-database-performance) to create additional filegroups and files for the BizTalk Server databases. This will greatly increase the performance of the BizTalk Server databases from a single disk configuration.
```
